x-pack/plugins/fleet/server/services/security/uninstall_token_service/index.ts

```diff
@@ -162,6 +162,12 @@ export interface UninstallTokenServiceInterface {
 export class UninstallTokenService implements UninstallTokenServiceInterface {
   private _soClient: SavedObjectsClientContract | undefined;
 
+  /** If BATCH_SIZE is too large, we get an error of `too_many_nested_clauses`.
+   *  Assuming that `max_clause_count` >= 1024, and experiencing that batch size should be less than half
+   *  than `max_clause_count` with our current query, batch size below 512 should be okay on every env.
+   */
+  private batchSize = 500;
```

Comment:
Should we make this value configurable in kibana.yml in case we need to override this for a customer?

Reply:
sure, that's the safest, added it: 5575dda5e4aba3a676c6b3ad616b270e74c3dc21

and maybe I should add a hint in case this error happens - will be back with that soon

Reply:
fe5297b1ce387ad2d7769242f5cde761d9cd027e
added the hint, and also made the error soft fail. actually, this makes sense: why should we block the user accessing fleet if tokens cannot be validated?

so now if there is an issue, it looks like the following without blocking fleet:

```
[2024-01-10T16:20:05.928+01:00][WARN ][plugins.fleet] Failed to validate uninstall tokens: `too_many_nested_clauses` error received. Setting/decreasing the value of `xpack.fleet.setup.uninstallTokenVerificationBatchSize` in your kibana.yml should help. Current value is 5000.: 'ResponseError: all shards failed: search_phase_execution_exception
	Caused by:
		too_many_nested_clauses: Query contains too many nested clauses; maxClauseCount is set to 6144
	Root causes:
		too_many_nested_clauses: Query contains too many nested clauses; maxClauseCount is set to 6144'
[2024-01-10T16:20:07.067+01:00][INFO ][plugins.ml] Task ML:saved-objects-sync-task: No ML saved objects in need of synchronization
[2024-01-10T16:20:07.137+01:00][INFO ][plugins.fleet] Encountered non fatal errors during Fleet setup
[2024-01-10T16:20:07.137+01:00][INFO ][plugins.fleet] {"name":"UninstallTokenError","message":"Failed to validate uninstall tokens: `too_many_nested_clauses` error received. Setting/decreasing the value of `xpack.fleet.setup.uninstallTokenVerificationBatchSize` in your kibana.yml should help. Current value is 5000."}
[2024-01-10T16:20:07.137+01:00][INFO ][plugins.fleet] Fleet setup completed
```

<img width="384" alt="image" src="https://github.com/elastic/kibana/assets/39014407/c7006166-d005-435f-aff4-68526977e67b">